libraries/AP_Param/AP_Param.cpp

```diff
@@ -2109,16 +2110,18 @@ bool AP_Param::parse_param_line(char *line, char **vname, float &value, bool &re
 // increments num_defaults for each default found in filename
 bool AP_Param::count_defaults_in_file(const char *filename, uint16_t &num_defaults)
 {
+    // try opening the file both in the posix filesystem and using AP::FS
     FILE *f = fopen(filename, "r");
-    if (f == nullptr) {
+    int file_apfs = AP::FS().open(filename, O_RDONLY);
+    if (f == nullptr && file_apfs == -1) {
         return false;
     }
     char line[100];
 
     /*
       work out how many parameter default structures to allocate
      */
-    while (fgets(line, sizeof(line)-1, f)) {
+    while ((f != nullptr &&fgets(line, sizeof(line)-1, f)) || AP::FS().fgets(line, sizeof(line)-1, file_apfs)) {
```

Comment:
should we want to check file_apfs here as well? i guess not necessary as it shouldn't get evaluated if f is valid.

```suggestion
    while ((f != nullptr &&fgets(line, sizeof(line)-1, f))|| (file_apfs != -1 && AP::FS().fgets(line, sizeof(line)-1, file_apfs))) {
```

Reply:
right. I did this under the (wrong) assumption that the second would only run if the first was an invalid path. but sofgzmething could go wrong in fgets, or something could make f nullptr again. updated. Thanks!